src/function_caller.jl

```diff
@@ -30,33 +31,27 @@ function (affect!::FunctionCallingAffect)(integrator, force_func = false)
         end
     end
     if affect!.func_everystep || force_func
-        affect!.funciter += 1
         affect!.func(integrator.u, integrator.t, integrator)
     end
     u_modified!(integrator, false)
 end
 
 function functioncalling_initialize(cb, u, t, integrator)
     funcat_cache = cb.affect!.funcat_cache
-    if cb.affect!.funciter != 0 || funcat_cache isa Number
+    if cb.affect!.funciter != 1 || funcat_cache isa Number
         tspan = integrator.sol.prob.tspan
-        funcat_cache = cb.affect!.funcat_cache
         funcat_vec = if funcat_cache isa Number
             step = funcat_cache
             t0, tf = tspan
             if !cb.affect!.func_start
                 t0 += step
             end
-            range(t0, tf; step)
-        else
-            funcat_cache
-        end
-        if integrator.tdir > 0
-            cb.affect!.funcat = BinaryMinHeap(funcat_vec)
+            integrator.tdir > 0 ? collect(t0:step:tf) : collect(reverse(t0:step:tf))
```

Comment:
does this actually matter? It sorts automtically.

Reply:
Do you mean that `tf < t0` if `tdir < 0`?

Reply:
yes

Reply:
Ok